### PR TITLE
add nightly docker build workflow

### PR DESCRIPTION
**Pull request by wilsonwang371**
add nightly docker build workflow

## Why are these changes needed?

add workflow for nightly operator build

## Related issue number

#70 

## Checks

- [x] I've made sure the tests are passing. 
- Testing Strategy
   - [ ] Unit tests
   - [X] Manual tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: .github/workflows/nightly.yaml [hunk 1]
```diff
@@ -0,0 +1,50 @@
+name: Nightly-build
+
+on:
+  schedule:
+    - cron: '0 2 * * *'
```
Comment (Jeffwan) [inline .github/workflows/nightly.yaml:5]: Can we make it event triggered? for example, only when we merge a commit?

File: .github/workflows/nightly.yaml [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    - name: Set up Docker
+      uses: docker-practice/actions-setup-docker@master
+
+    - name: install kubebuilder
+      run: |
+        wget https://github.com/kubernetes-sigs/kubebuilder/releases/download/v3.0.0/kubebuilder_$(go env GOOS)_$(go env GOARCH)
+        sudo mv kubebuilder_$(go env GOOS)_$(go env GOARCH) /usr/local/bin/kubebuilder
+
+    - name: Build Docker Image
+      run: IMG=kuberay/operator:nightly make docker-image
```
Comment (Jeffwan) [inline .github/workflows/nightly.yaml:40]: I kind of forget whether we need kubebuilder to make docker-image?

File: .github/workflows/nightly.yaml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+      working-directory: ${{env.working-directory}}
+    
+    - name: Log in to Docker Hub
+      uses: docker/login-action@f054a8b539a109f9f41c372932f1ae047eff08c9
+      with:
+        username: ${{ secrets.DOCKER_USERNAME }}
```
Comment (Jeffwan) [inline .github/workflows/nightly.yaml:46]: We need help from community member who has repo permission to configure it, right?

Comment (wilsonwang371) [reply .github/workflows/nightly.yaml:46]: Yes, if this is not a convenient way, we need to think of a different apporach.

File: .github/workflows/nightly.yaml (hunk 1 above)
Comment (wilsonwang371) [reply .github/workflows/nightly.yaml:5]: Do you mean we run nightly build on event-based(commit to main branch)?

File: .github/workflows/nightly.yaml (hunk 2 above)
Comment (wilsonwang371) [reply .github/workflows/nightly.yaml:40]: I think we can remove kubebuilder here.

File: .github/workflows/test-job.yaml [hunk 4]
```diff
@@ -106,9 +106,21 @@ jobs:
       uses: docker-practice/actions-setup-docker@master
 
     - name: Build Docker Image
-      run: make docker-image
+      run: IMG=kuberay/operator:nightly make docker-image
```
Comment (Jeffwan) [inline .github/workflows/test-job.yaml:109]: @wilsonwang371 one comment from my side.

I suggest to use commit id as the tag and it's easier to target the problem. 

1. for PR changes, let's use local image and load into kind for testing but skip pushing to registry
2. for merge change, we can push to registry and retag to nightly and push it again.

**Review by Jeffwan — APPROVED** [review]

**Comment by Jeffwan** [comment]
Great. With this change, we should be able to upload latest container image to dockerhub for every commit

**Pull request merged**